### ISSUE TITLE
Unexpected micronutrient changes after modelling reduced yields

### ISSUE THREAD

**Issue by celiapetty** [dbrownukk/EFD_v2#640]
Instance QA
Module: OIHM livelihoods modelling
Study :Katakwi - NIMFRU - AIHM 2018
Title: Katakwi AIHM match
Reduced crop yields as shown to 1
Micronutrients appear to have increased after the reduction in food income. I would expect these to decrease.
Spreadsheet attached.
Any thoughts?
[Change2NIMFRUThu May 13 17_32_25 UTC 2021.xlsx](https://github.com/dbrownukk/EFD_v2/files/6474137/Change2NIMFRUThu.May.13.17_32_25.UTC.2021.xlsx)

**Comment by daiClegg** [comment]
After that change they are buying the default diet to make up for the lost food income. 
You can see this because the DI goes down but the CI doesn't

Might it be that the default diet is more nutritious than the crops they've lost?
I'm not convinced, but if that were the case you would expect to see results like this wouldn't you?

**Comment by celiapetty** [comment]
I'm not sure-the default diet is sweet potato, sorghum and cassava-all 3 are included in the reduced yield model.

**Comment by daiClegg** [comment]
In that case, is there one HH that doesn't rely on much else apart from the crops in the default diet and just model the impact on that one. If i'm right there should be only a small change in nutrition levels.
If there's still a significant delta it looks like a bug.

The thing i'd be looking for would be which nutrients are most markedly different and see if their relative values are in the changed diet would account for the differences.

**Comment by celiapetty** [comment]
Will take a look

**Comment by celiapetty** [comment]
Just scanned through each hh's crops. Closest is hh 79 which only produces SwP and Sorghum.  Their change scenario looks similar to the others.  I also noted that the change scenario lists 40 hhs whereas the study module has 42 hhs.  There is a rogue household (Dave) hh 92 -in the change scenario modelling list. Did this get included in testing?  Finally, HH 1 is among those missing in the change scenario report. This hh also had limited crops (just cassava and sorghum) no change scenario available.  Unless there is an obvious explanation for the missing households in the change scenario, I will add this as an issue.

**Comment by daiClegg** [comment]
HH1 & HH5 are not validated so they are excluded from reporting and modelling.
HH92 is a test HH; bear in mind this isn't the production data it is a test data set. 
Feel free to delete that HH if it's distorting results.

**Comment by celiapetty** [comment]
Thanks for this clarification. I'll check HH1 and 5 and will delete the
test hhold

**Comment by celiapetty** [comment]
I've just re-run katakwi-nimfru AIHM 2018
created drb 7.10.2020

Change scenario
Cassava,   dry Yield = 3% Price = 100%
--
Groundnuts, dry Yield = 3%   Price = 200%
Maize, dry Yield = 5% Price =   300%
Rice Yield = 5% Price = 200%
Sweet Potato, dry Yield = 5%   Price = 300
Results still don't look right: Micronutrients go up by what looks like 50% more or less consistently. Could it be an error that assumes same amount of micronutrients concentrated in lower yields??? Probably a bit far fetched!

**Comment by daiClegg** [comment]
From an offline discussion:

The Micronutrients report was intended to show the resulting change in nutrient levels after a Food Substitution Change Scenario. So neither Price Yield changes nor Coping Strategies are taken into account. 
That needs further discussion

However, in running a very basic example with no Food Substitution the nutrient levels after 'Food Change Scenario' (on the report) appear to double.
To be investigated.

**Comment by dbrownukk** [comment]
Yes, there was an error with no food substitution. Fixed and deployed to TESTING

A Food item cannot be used both in Price Yield Variation and Food Substitution - a restriction is built in.

If you remove Groundnuts, dry from Price Yield Variation and add a Food substitution from Ground Nuts, dry to Chick Peas dry (not sure if that would make sense, just an example) you will see a change in the micronutrient levels.

**Comment by daiClegg** [comment]
@dbrownukk Is this ready for testing? It's not labelled as such??

**Comment by dbrownukk** [comment]
yes, sorry missed updating status..

**Comment by daiClegg** [comment]
INstance: Testing
App: OIHM/Modelling
Scenario: Test Micronutrients
Project: NIMFRU
Study:Katakwi - NIMFRU - AIHM 2018
HH:79

This HH receives and consumes 900 Kgs of Mangoes.
They do not grow (crop) or forage (Wild) Mangoes

The change Scenario is 50% drop in yield for Mango (as Crop) Mango (as Wild Food) and Mango (as Food Transfer)
(there was also a crop substitution in the scenario as i first ran it, but that was not relevant to this HH)

1. In Modeling Menu, select Modelling
2. Select Scenario as above
3. Click Reports
4. Select HH as above
5. Click Run
6. Open the downloaded Data Extract

> Attached annotated:
> [ChangeNIMFRUWed Jul 21 09_29_55 UTC 2021.xlsx](https://github.com/dbrownukk/EFD_v2/files/6855179/ChangeNIMFRUWed.Jul.21.09_29_55.UTC.2021.xlsx)

7. Check Food Income report tab
> note the food income loss from the reduced transfer is correctly reported 
8. Check Nutrition report tab
> note the scenario has no impact on nutrition; it should show a significant diminution of all nutrients present in Mangoes

**Comment by dbrownukk** [comment]
Definitely something amiss - checking priceyield and substitution calc for micronutrients
However, Mango is not mapped to a micronutrient in MccW food source, thus Mango will never have any nutrient effect. But, think the calc is missing yield change when reviewing code.

**Comment by daiClegg** [comment]
Had not spotted that Mango is unmapped. Bit silly trying to create  minimal case and then leaving a hole in it.
Sorry.

**Comment by dbrownukk** [comment]
Back to comment of 21 June:
_The Micronutrients report was intended to show the resulting change in nutrient levels after a Food Substitution Change Scenario. So neither Price Yield changes nor Coping Strategies are taken into account.
That needs further discussion_      (I thought I had lost the nutrient yield change code...)

The levels are definitely wrong, but the 50% yield change is a bad coincidence, in that the values were being doubled, but has nothing to do with Price Yield Changes.

**Comment by dbrownukk** [comment]
Nutrient report with Substitution Food changes now appears correct. Note that due to the new drop down Units for assets in the application a postload trigger is firing. This is OK for HH reports but seems to have slowed the initial selection of Villages for OHEA when running the modelling report. The generation of the excel is still pretty fast. Will raise new log to look into how the postload trigger can be avoided, but will need a new strategy for drop down Units...

deployed to IDAPS_livelihoods_testing
